### DISCUSSION TITLE
NextJS & Vercel _next fallback rewrites

### DISCUSSION THREAD

**Discussion by sambennett114** [vercel/community#5536] [Help]
### Summary

Hello! I am trying to setup rewrites for two NextJS websites. One is hosted on Vercel and another is hosted on a different provider (AWS).

I have a fallback rewriting everything to the separate domain and it works locally. When deployed to Vercel, requests to `/_next/**` are not falling back to this rewrite causing broken pages on the rewritten pages.

Is this expected behavior and is there a workaround for it? I've also tried using middleware, but the requests are a lot harder to manage especially for prefetching, etc. and would prefer to have a fallback so we don't have to account for every single route.

```
  rewrites: async () => {
    return {
      beforeFiles: [],
      afterFiles: [],
      fallback: [
        {
          source: "/:file(.*\\.(?:\\w+))",
          destination: `https://www.domain.com/:file*`,
        },
        {
          source: "/:path*",
          destination: `https://www.domain.com/:path*/`,
        },
      ],
    };
  },
```

I have also tried something like this and it still did not work:
```
  rewrites: async () => {
    return {
      beforeFiles: [],
      afterFiles: [],
      fallback: [
        {
          source: "/_next/:file*",
          destination: `https://www.domain.com/_next/:file*`,
        },
        {
          source: "/:path*",
          destination: `https://www.domain.com/:path*/`,
        },
      ],
    };
  },
```

### Steps to Reproduce

You can use the above rewrite configuration in any nextjs project.

Run the project locally and all the rewrites happen correctly.

Deploy to Vercel and the fallback for _next does not work.

**Comment by oldbrush** [comment]
@sambennett114 Using https://nextjs.org/docs/pages/api-reference/next-config-js/assetPrefix may be a better approach as it will redirect .next/static contents to the AWS-hosted website. Please let me know if you have any further questions.

**Reply by sambennett114** [reply]
@oldbrush Thanks! I wasn't sure if assetPrefix would work or not but will definitely test. Will follow up here with our findings
